### DISCUSSION TITLE
Change Hyperlink Color?

### DISCUSSION THREAD

**Discussion by stigzler** [icsharpcode/AvalonEdit#427] [Q&A] [answered]
Hi folks - I'm converting all the default xshd files for DarkMode applications. On stumbling block I've hit is how to change the colour of hyperlinks. At present they look like this:

![Screenshot 2024-06-08 184016](https://github.com/icsharpcode/AvalonEdit/assets/4856029/6bd0c04a-ff3c-4e2c-8a3b-6ddc8b19779b)

Ideally, the solution would be changing this via the xshd file, but I'm not sure this is possible? Second best would be how to change this programmatically and globally for the editor when switching to dark mode, if such a feature exists. Done a bit of a search - can't find anything on this....

Hope someone can help. Loving this btw 👍

**Comment by ltrzesniewski** [comment, marked as answer]
> changing this via the xshd file, but I'm not sure this is possible

That's not possible, unfortunately. The hyperlink generator uses a different mechanism than highlighting.

> how to change this programmatically

You can set the default color like this, by having a default style for `TextView` in scope:

```xml
<Style TargetType="{x:Type avalonedit:TextView}">
    <Setter Property="LinkTextForegroundBrush" Value="#99DAF9" />
</Style>
```

Or by changing that property from C#:

```csharp
yourTextEditor.TextArea.TextView.LinkTextForegroundBrush = Brushes.LightBlue;
```

**Reply by stigzler** [reply]
Awesome sauce. Cheers shipmate. 👍
